### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by fcooper8472**
First draft of unit testing course event page.

Running a bit short on time, so reviews/suggestions ASAP if possible please!

### CODE REVIEW

File: content/events/2021-02-17_unit_testing_course.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+We will use a small Python project to introduce concepts and to get some hands-on experience writing unit tests, and we will use GitHub to automate the process of running them.
+
+Participants should have a [GitHub](http://github.com/) account and have a basic familiarity with Python or a similar programming language. 
+
+This course will cover:
+- what is software testing
+- using a unit testing framework
+- writing unit tests
+- testing for errors
+- parameterising unit tests
```
Comment (ben18785) [inline content/events/2021-02-17_unit_testing_course.md:32]: Worth explaining what "parameterising" means to a layman? (Like me.) If the aim is an advert, I suspect it'd help to put it in language that non-software people could more easily understand.

File: content/events/2021-02-17_unit_testing_course.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+The Oxford Research Software Engineering group is running a free half day course on unit testing for graduate students and researchers at the University of Oxford.
+
+
+## Overview
+
+Does the code we develop work the way we think it does?
+Can we verify the correctness of our code?
+To what extent do we have confidence in the accuracy of results that we publish?
```
Comment (ben18785) [inline content/events/2021-02-17_unit_testing_course.md:20]: Worth saying something about "When you extend your code / when you work collectively on code as part of a group, how can you be sure that previously working functionality continues to do so?".

**Review by ben18785 — COMMENTED** [review]
Could of comments: mainly from a relative layman's perspective!

File: content/events/2021-02-17_unit_testing_course.md (hunk 2 above)
Comment (tlestang) [reply content/events/2021-02-17_unit_testing_course.md:20]: I agree. Two points that may be relevant to researchers may be:
- Automated tests (software tested by software)
- Tests mean less debugging time

File: content/events/2021-02-17_unit_testing_course.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+17 February 2021 ~ Remote (Zoom) ~ Sign-up on Eventbrite coming soon
+
+
+The Oxford Research Software Engineering group is running a free half day course on unit testing for graduate students and researchers at the University of Oxford.
+
+
+## Overview
+
+Does the code we develop work the way we think it does?
```
Comment (tlestang) [inline content/events/2021-02-17_unit_testing_course.md:18]: Isn't that the same at _Can we verify the correctness of our code?_

**Review by tlestang — COMMENTED** [review]
Thanks for putting this together! Do you think it's worth introducing CI with GitHub on top of unit testing?

**Comment by fcooper8472** [comment]
> Thanks for putting this together! Do you think it's worth introducing CI with GitHub on top of unit testing?

My plan was to have the CI already running: have the github actions script there right from the start.  Then that part would be "now we have tests it's really annoying to keep running them manually, but if you look over here on GitHub they're already being run for you!"

**Pull request merged**